lib/Parse/Lexer.cpp

```diff
@@ -827,10 +827,15 @@ void Lexer::lexDollarIdent() {
     }
   }
 
-  // It's always an error to see a standalone $, and we reserve
-  // $nonNumeric for persistent bindings in the debugger.
-  if (CurPtr == tokStart + 1 || !isAllDigits) {
-    if (!isAllDigits && !LangOpts.EnableDollarIdentifiers)
+  // It's always an error to see a standalone $
+  if (CurPtr == tokStart + 1) {
+    diagnose(tokStart, diag::expected_dollar_numeric);
+    return formToken(tok::unknown, TokStart);
+  }
+
+  // We reserve $nonNumeric for persistent bindings in the debugger.
+  if(!isAllDigits) {
+    if (!LangOpts.EnableDollarIdentifiers)
```

Comment:
Did you intend to include the config changes that have to go along with this flag?    You'll have to do that in the `.td` files not in the compiled config files.